packages/app/src/cli/services/environment/id-matching.ts

```diff
@@ -143,5 +145,33 @@ export async function automaticMatchmaking(
     toConfirm,
     toCreate,
     toManualMatch: pending,
+    toMigrate: getExtensionsToMigrate(localSources, remoteSources, ids),
   }
 }
+
+function getExtensionsToMigrate(
+  localSources: LocalSource[],
+  remoteSources: RemoteSource[],
+  ids: {[key: string]: string},
+) {
+  const remoteExtensionTypesToMigrate = ['CHECKOUT_UI_EXTENSION']
+
+  return localSources.reduce<MatchResult['toMigrate']>((accumulator, localSource) => {
+    if (localSource.type === 'ui_extension') {
+      const remoteSource = remoteSources.find((source) => source.uuid === ids[localSource.configuration.name])
```

Comment:
This logic assumes you have the UUID of the extension locally right? that won't be always true.
The logic should be similar to `matchByNameAndType` but instead of matching by same type we match by `remoteExtensionTypesToMigrate` and `ui_extension`

In any case, adding a bunch of tests for different scenarios would help us detect any possible not covered edge case.

We can pair next week on this if you need to (or also maybe @Arkham can help, since he implemented most of this logic 😃 )

Reply:
Does the matching by name logic account for the local name not matching the remote source's name? I think currently this can get out of date as developers can change the name through the Partner's dashboard. Do we then treat it as a different extension even though the uuid would match?

Reply:
I'm going to pair with @Arkham on this tomorrow, so will check this logic.